src/xb-database-manager.c

```diff
@@ -658,8 +650,26 @@ xb_database_manager_query (XbDatabaseManager *self,
       goto out;
     }
 
-  parsed_query = xapian_query_parser_parse_query_full (payload->qp, query_str,
-                                                       QUERY_PARSER_FLAGS, "", &error);
+  const gchar *match_all;
+  match_all = g_hash_table_lookup (query_options, QUERY_PARAM_MATCH_ALL);
+  if (match_all != NULL && str == NULL)
+    {
+      parsed_query = xapian_query_new_match_all ();
+    }
+  else if (str != NULL && match_all == NULL)
+    {
+      /* save the query string aside */
+      query_str = g_strdup (str);
+      parsed_query = xapian_query_parser_parse_query_full (payload->qp, query_str,
+                                                           QUERY_PARSER_FLAGS, "", &error);
+    }
+  else
+    {
+      g_set_error(error_out, XB_ERROR,
+                  XB_ERROR_INVALID_PARAMS,
+                  "Exactly one of query parameter or match all parameter is required.");
```

Comment:
Wording is a bit unnatural here; how about "Either 'query' or 'matchAll' parameter is required."?